### ISSUE TITLE
Issue with sklearn and skopt. Cannot import name 'MaskedArray' again

### ISSUE THREAD

**Issue by adamhall** [scikit-optimize/scikit-optimize#1077]
I was asked to open a new issue for this in #1061, and seems to be the same issue as #902.
I created a virtual env using: 
`virtualenv venv`

This installs python 3.8.10. Then, running
`
pip3 install scikit-optimize
`
installs 
```
numpy           1.21.2
scikit-learn    1.0
scikit-optimize 0.8.1
scipy           1.7.1
```
Simply calling

`from skopt.sampler Lhs`

Gives the error
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ahall/Documents/projects/skopt_test/venv/lib/python3.8/site-packages/skopt/__init__.py", line 55, in <module>
    from .searchcv import BayesSearchCV
  File "/home/ahall/Documents/projects/skopt_test/venv/lib/python3.8/site-packages/skopt/searchcv.py", line 16, in <module>
    from sklearn.utils.fixes import MaskedArray
ImportError: cannot import name 'MaskedArray' from 'sklearn.utils.fixes' (/home/ahall/Documents/projects/skopt_test/venv/lib/python3.8/site-packages/sklearn/utils/fixes.py)
```

**Comment by adamhall** [comment]
should 
https://github.com/scikit-optimize/scikit-optimize/blob/708bf1645c9767beafd4b198eb2ee5935dffc8ff/pyproject.toml#L8
be changed to somwthing like (I don't know what the syntax is for packaging)
`"scikit-learn>=0.19.1, <0.24.2"`?

**Comment by kernc** [comment]
Assumed fixed by v0.9.0 just released.
